dbt/adapters/redshift/impl.py

```diff
@@ -99,3 +104,53 @@ def valid_incremental_strategies(self):
 
     def timestamp_add_sql(self, add_to: str, number: int = 1, interval: str = "hour") -> str:
         return f"{add_to} + interval '{number} {interval}'"
+
+    def _link_cached_database_relations(self, schemas: Set[str]):
+        """
+        :param schemas: The set of schemas that should have links added.
+        """
+        database = self.config.credentials.database
+        _Relation = namedtuple("_Relation", "database schema identifier")
+        links = [
+            (
+                _Relation(database, dep_schema, dep_identifier),
+                _Relation(database, ref_schema, ref_identifier),
+            )
+            for dep_schema, dep_identifier, ref_schema, ref_identifier in self.execute_macro(
+                GET_RELATIONS_MACRO_NAME
+            )
+            # don't record in cache if this relation isn't in a relevant schema
+            if ref_schema in schemas
+        ]
+
+        for dependent, referenced in links:
+            self.cache.add_link(
+                referenced=self.Relation.create(**referenced._asdict()),
+                dependent=self.Relation.create(**dependent._asdict()),
+            )
+
+    def _link_cached_relations(self, manifest):
+        schemas = set(
+            relation.schema.lower()
+            for relation in self._get_cache_schemas(manifest)
+            if self.verify_database(relation.database) == ""
+        )
+        self._link_cached_database_relations(schemas)
+
+    def _relations_cache_for_schemas(self, manifest, cache_schemas=None):
+        super()._relations_cache_for_schemas(manifest, cache_schemas)
+        self._link_cached_relations(manifest)
+
+    # avoid non-implemented abstract methods warning
+    # make it clear what needs to be implemented while still raising the error in super()
+    # we can update these with Redshift-specific messages if needed
+    @property
+    def python_submission_helpers(self) -> Dict[str, Type[PythonJobHelper]]:
```

Comment:
These are technically still abstract methods, eh? Since they super to unimplemented ones under the hood?